Создание API. Postman Домашняя работа от 07/02/2023

ruslankorotkov opened the pull request:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения
RecipesApp - приложение для сайта рецептов.
06/02/2023

Commit:
Структура веб-приложения
RecipesApp - приложение для сайта рецептов.
06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 06/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения.
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 07/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод c/reateIngredient", summary = "Можете ввести информацию об новом ингредиент", description = "Можно ввести информацию")
+    @PostMapping("/createIngredient")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. POST - создать/добавить, путь должен быть просто слэшем

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /getIngredient/{id}", summary = "Можете получить информацию об ингредиенте по id", description = "Можно получить информацию")
+    @GetMapping("/getIngredient/{Id}")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. GET - получить, путь должен быть просто слэшем и id

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /updateIngredient", summary = "Можете изменить информацию об ингредиенте по параметрам", description = "Можно изменить информацию")
+    @PutMapping("/updateIngredient")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. PUT - обновить, путь должен быть просто слэшем

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /deleteIngredient/{id}", summary = "Можете удалить информацию об ингредиенте по id", description = "Можно удалить информацию")
+    @DeleteMapping("/deleteIngredient/{Id}")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. DELETE - удалить, путь должен быть просто слэшем и id

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /deleteIngredient/{id}", summary = "Можете удалить информацию об ингредиенте по id", description = "Можно удалить информацию")
+    @DeleteMapping("/deleteIngredient/{Id}")
+    public ResponseEntity deleteIngredient(@RequestParam Long id) {
```
аннотация неверная, у тебя параметр пути, а не запроса

dmitry-bizin replied:
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /updateIngredient", summary = "Можете изменить информацию об ингредиенте по параметрам", description = "Можно изменить информацию")
+    @PutMapping("/updateIngredient")
+    public ResponseEntity updateIngredient(@RequestParam Long id, @RequestParam Ingredient ingredient) {
```
аннотация неверная, у тебя параметр пути, а не запроса

dmitry-bizin replied:
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /getAllIngredients", summary = "Можете получить информацию о всех ингредиентах", description = "Можно получить информацию")
+    @GetMapping("/allIngredients")
+    public ResponseEntity getAllIngredients(@PathVariable Map<Long, Ingredient> ingredientMap) {
```
аннотация неверная, у тебя параметры запроса, а не пути

dmitry-bizin replied:
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /getAllIngredients", summary = "Можете получить информацию о всех ингредиентах", description = "Можно получить информацию")
+    @GetMapping("/allIngredients")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. GET - получить, путь должен быть просто слэшем

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /getIngredient/{id}", summary = "Можете получить информацию об ингредиенте по id", description = "Можно получить информацию")
+    @GetMapping("/getIngredient/{Id}")
+    public ResponseEntity getIngredient(@PathVariable Long id) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/IngredientController.java:
```diff
@@ -0,0 +1,69 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод c/reateIngredient", summary = "Можете ввести информацию об новом ингредиент", description = "Можно ввести информацию")
+    @PostMapping("/createIngredient")
+    public ResponseEntity createIngredient(@RequestBody Ingredient ingredient) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод c/createRecipe", summary = "Можете ввести информацию о новом рецепте", description = "Можно ввести информацию")
+    @PostMapping("/createRecipe")
+    public ResponseEntity createRecipe(@RequestBody Recipe recipe) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /getRecipe/{id}", summary = "Можете получить информацию о рецепте по id", description = "Можно получить информацию")
+    @GetMapping("/getRecipe/{id}")
+    public ResponseEntity getRecipe(@PathVariable Long id) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /updateRecipe", summary = "Можете изменить информацию  о рецепте по параметрам", description = "Можно изменить информацию")
+    @PutMapping("/updateRecipe/{id}")
+    public ResponseEntity updateRecipe(@RequestBody Long id, @RequestBody Recipe recipe) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin replied:
аннотация неверная для параметра пути

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /deleteRecipe//{id}", summary = "Можете удалить информацию  о рецепте по id", description = "Можно удалить информацию")
+    @DeleteMapping("/deleteRecipe/{id}")
+    public ResponseEntity deleteRecipe(@RequestBody Long id) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin replied:
аннотация неверная для параметра пути

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /allRecipes", summary = "Можете получить информацию о всех рецептах", description = "Можно получить информацию")
+    @GetMapping("/allRecipes")
+    public ResponseEntity getAllRecipes(@PathVariable Map<Long, Recipe> recipeMap) {
```
ResponseEntity надо использовать с дженериками

dmitry-bizin replied:
аннотация неверная, у тебя параметры запроса, а не пути

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод c/createRecipe", summary = "Можете ввести информацию о новом рецепте", description = "Можно ввести информацию")
+    @PostMapping("/createRecipe")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. POST - создать/добавить, путь должен быть просто слэшем

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /getRecipe/{id}", summary = "Можете получить информацию о рецепте по id", description = "Можно получить информацию")
+    @GetMapping("/getRecipe/{id}")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. GET - получить, путь должен быть просто слэшем и id

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /updateRecipe", summary = "Можете изменить информацию  о рецепте по параметрам", description = "Можно изменить информацию")
+    @PutMapping("/updateRecipe/{id}")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. PUT - обновить, путь должен быть просто слэшем

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /allRecipes", summary = "Можете получить информацию о всех рецептах", description = "Можно получить информацию")
+    @GetMapping("/allRecipes")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. GET - получить, путь должен быть просто слэшем

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/controllers/RecipesController.java:
```diff
@@ -0,0 +1,67 @@
+
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Информация была получена", content = {@Content(mediaType = "application/json")})})
+    @Operation(method = "Метод /deleteRecipe//{id}", summary = "Можете удалить информацию  о рецепте по id", description = "Можно удалить информацию")
+    @DeleteMapping("/deleteRecipe/{id}")
```
здесь неверно, нарушен принцип REST. В путях не должно быть глаголов, сам метод уже говорит о действии. DELETE - удалить, путь должен быть просто слэшем и id

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/model/Recipe.java:
```diff
@@ -0,0 +1,16 @@
+public class Recipe {
+    private String title;
+    private int cookingTime;
+    private Map<Long, Ingredient> ingredientMap;
```
Здесь должен быть список, а не мапа (см. задание)

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/model/Recipe.java:
```diff
@@ -0,0 +1,16 @@
+    private String title;
+    private int cookingTime;
+    private Map<Long, Ingredient> ingredientMap;
+    private Map<Long, String> cookingInstructionsSteps;
```
Здесь должен быть список, а не мапа (см. задание)

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/IngredientServiceimpl.java:
```diff
@@ -0,0 +1,44 @@
+@Service
+public class IngredientServiceimpl implements IngredientService {
+
+    private Map<Long, Ingredient> ingredientMap = new LinkedHashMap<>();
```
final

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/IngredientServiceimpl.java:
```diff
@@ -0,0 +1,44 @@
+public class IngredientServiceimpl implements IngredientService {
+
+    private Map<Long, Ingredient> ingredientMap = new LinkedHashMap<>();
+    private Long generatedId = 1L;
```
long

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/IngredientServiceimpl.java:
```diff
@@ -0,0 +1,44 @@
+
+    @Override
+    public Ingredient updateIngredient(Long id, Ingredient ingredient) {
+        ingredientMap.put(id, ingredient);
```
здесь неверно, надо проверять, есть ли такой ингредиент и только потом обновлять, иначе вернуть null

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/IngredientServiceimpl.java:
```diff
@@ -0,0 +1,44 @@
+
+    @Override
+    public Ingredient deleteIngredient(Long id) {
+        return ingredientMap.remove(id);
```
здесь неверно, надо проверять, есть ли такой ингредиент и только потом удалять, иначе вернуть null

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/RecipesServiceimpl.java:
```diff
@@ -0,0 +1,47 @@
+public class RecipesServiceimpl implements RecipesService {
+
+
+    private Map<Long, Recipe> recipesMap = new LinkedHashMap<>();
```
final

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/RecipesServiceimpl.java:
```diff
@@ -0,0 +1,47 @@
+
+
+    private Map<Long, Recipe> recipesMap = new LinkedHashMap<>();
+    private Long generatedId = 1L;
```
long

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/RecipesServiceimpl.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public Recipe updateRecipe(long id, Recipe recipe) {
+        recipesMap.put(id, recipe);
```
здесь неверно, надо проверять, есть ли такой ингредиент и только потом обновлять, иначе вернуть null

dmitry-bizin commented on src/main/java/sky/pro/recipesapp/services/impl/RecipesServiceimpl.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public Recipe deleteRecipe(Long id) {
+        return recipesMap.remove(id);
```
здесь неверно, надо проверять, есть ли такой ингредиент и только потом удалять, иначе вернуть null

Commit:
Структура веб-приложения.
Создание API. Postman
Подключение библиотек
RecipesApp - приложение для сайта рецептов.
Домашняя работа от 08/02/2023
